### PR TITLE
[WIP] change sql query to active record

### PR DESCRIPTION
**Pull request by rorong**

### CODE REVIEW

File: app/models/account.rb [hunk 1]
```diff
@@ -153,10 +153,7 @@ def trigger
   end
 
   def change_balance_and_locked(delta_b, delta_l)
-    self.balance += delta_b
-    self.locked  += delta_l
-    self.class.connection.execute "update accounts set balance = balance + #{delta_b}, locked = locked + #{delta_l} where id = #{id}"
-    add_to_transaction # so after_commit will be triggered
+    self.update_attributes(balance: self.balance + delta_b, locked: self.locked + delta_l)
```
Comment (yivo) [inline app/models/account.rb:156]: @rorong I am very curious about transaction in this example? Can you add comments and check if everything works well? Test manually pls.

File: spec/models/account_spec.rb [hunk 2]
```diff
@@ -245,7 +245,7 @@
 
     it 'should retry the whole transaction on stale object error' do
       # `unlock_and_sub_funds('5.0'.to_d, locked: '8.0'.to_d, fee: ZERO)`
-      ActiveRecord::Base.connection.execute "update accounts set balance = balance + 3, locked = locked - 8 where id = #{subject.id}"
+      subject.update_attributes(balance: subject.balance + 3, locked: subject.locked - 8)
```
Comment (yivo) [inline spec/models/account_spec.rb:248]: @rorong For this example `update_attributes` works differently compared to plain SQL. Pls find out why.

**Review by yivo — CHANGES_REQUESTED** [review]

**Comment by yivo** [comment]
@rorong What this PR is related to?

File: spec/models/account_spec.rb (hunk 2 above)
Comment (rorong) [reply spec/models/account_spec.rb:248]: @yivo I am working on it

File: app/models/account.rb (hunk 1 above)
Comment (rorong) [reply app/models/account.rb:156]: @yivo ok, i need to test it as 1 test is failing i need to find why and solve it

Comment (yivo) [reply app/models/account.rb:156]: @rorong We don't good test coverage now. Pls consider testing manually deposits / withdraws / trading.

Comment (rorong) [reply app/models/account.rb:156]: ok

**Comment by yivo** [comment]
@rorong Hello. Please, describe what this PR is related to? Also what is the progress on it?

**Comment by rorong** [comment]
@yivo 
This is related to https://github.com/rubykube/peatio/issues/207

**Comment by rorong** [comment]
@yivo 
@calj Tell me to convert raw sql to active record but i am facing some issue init.

**Comment by rorong** [comment]
@yivo can you please tell me when this event is fired

**Comment by yivo** [comment]
@rorong Sorry, what event?

**Comment by rorong** [comment]
@yivo Means when i do manually check then from where this actions is called.

File: spec/models/account_spec.rb (hunk 2 above)
Comment (calj) [reply spec/models/account_spec.rb:248]: Anyway we can't make the sum in ruby, we need this code to be [reentrant](https://en.wikipedia.org/wiki/Reentrancy_(computing)).

When the sum is made in SQL it's applied implicitly in a transaction, so if 2 concurrents threads try to increase the balance the final balance will be correct.

**Comment by yivo** [comment]
@rorong I believe you should "grep" whole project for the method name. This is the one solution for finding usaged. The second other will be digging into source code details. I think you can also play with Peatio, get everything running and check if deposit / withdraw / trading work as expected.

**Comment by rorong** [comment]
@yivo Ok, I have to play around it to check its functionality

File: spec/models/account_spec.rb (hunk 2 above)
Comment (rorong) [reply spec/models/account_spec.rb:248]: @calj ok

**Comment by rorong** [comment]
@yivo @calj 

I've tried to do in active record but got the some unexpected behaviour of adding two transaction value with active record. I thoughts, we have to insatlled the MSSQL server and write the code and test the flow.

**Comment by yivo** [comment]
@rorong Will you install MSSQL and test it?

**Comment by rorong** [comment]
@yivo I will install MSSQL and test it.

**Comment by yivo** [comment]
@rorong Can you finish this issue till Friday?

**Comment by rorong** [comment]
@yivo I have installed the MSSQL server i will try to finish it till Friday

**Pull request closed without merging**